platforms/platform-googlebusiness/src/GoogleBusinessPlatform.ts

```diff
@@ -97,6 +103,32 @@ export class GoogleBusinessPlatform extends Platform<
     return response;
   }
 
+  private async handlePotentialDuplicateMessage(jovo: Jovo) {
+    if (!jovo.$googleBusiness) {
+      return;
+    }
+    const messageId =
+      jovo.$googleBusiness.$request.message?.messageId ||
+      jovo.$googleBusiness.$request.suggestionResponse?.message?.match?.(/messages[/](.*)/)?.[1];
+    if (!messageId) {
+      return;
+    }
+    const processedMessages = jovo.$session.data._GOOGLE_BUSINESS_PROCESSED_MESSAGES_ || [];
+    // message was handled already
+    if (processedMessages.includes(messageId)) {
+      jovo.$handleRequest.stopMiddlewareExecution();
+      return jovo.$handleRequest.server.setResponse({});
+    }
+    processedMessages.push(messageId);
+    jovo.$session.data._GOOGLE_BUSINESS_PROCESSED_MESSAGES_ = processedMessages;
+
+    const dbPlugins = Object.values(jovo.$handleRequest.plugins).filter(
```

Comment:
Quick question for me to understand this: I was under the assumption that the session data is stored to the DB anyways (happening later in the lifecycle). Could you tell me why we need to access the DB plugin here? Thanks!

Reply:
Like described in this [commit](https://github.com/jovotech/jovo-framework/commit/e6ad0633d14574ecb5b5a0468c87d304f03f0c68):

> An immediate save is required because a Lambda may start multiple instances which then need to be aware of the request from the other instance.

Basically, we want other instances that start a little bit delayed to be able to check if the message was handled or is being handled. This is an edge-case probably but should be taken in consideration.

Reply:
Cool, thanks. Could you add this as a comment?